File: IntroClasses/mago.cpp
#include <iostream>
using namespace std;

class Mago {
public:
    // Atributos de classe
    static bool possuiMagia;

    // Método construtor
    Mago(string nome, int idade, string escola) {
        // Atributos de instância
        this->nome = nome;
        this->idade = idade;
        this->escola = escola;
        cout << "Mago " << this->nome << " foi criado!" << endl;
    }

    // Outros métodos
    void andar() {
        cout << "Estou andando" << endl;
    }

    void falar() {
        cout << "Ola amigue! Meu nome é " << this->nome << endl;
    }

    void invocarMagia() {
        cout << "Invocando magia!" << endl;
    }

    // Destrutor
    ~Mago() {
        cout << "Deixou de existir!" << endl;
    }

private:
    string nome;
    int idade;
    string escola;
};

bool Mago::possuiMagia = true;

int main() {
    Mago mago1("Gandalf", 150, "Aprendizes de Merlin");
    mago1.andar();
    mago1.falar();
    mago1.invocarMagia();

    return 0;
}

File: IntroClasses/mago.py
# Definição da classe Mago 

class Mago:
    # Atributos de classe
    possuiMagia = True

    # Método construtor
    def __init__(self, nome, idade, escola):
        # Atributos de instância
        self.__nome = nome 
        self.idade = idade   
        self.escola = escola 
        print('Mago ', self.__nome, ' foi criado!')

    # Outros métodos    
    def andar(self):
        print('Estou andando')
    
    def falar(self):
        print('Ola amigue! Meu nome é ',self.__nome)
        
    def invocarMagia(self):
        print('Invocando magia!')

    def cumprimentar(self,nome):
        print('Ola, ', nome)

    # Método destrutor
    def __del__(self):  
        print('Deixou de existir!') 

    #Método de set 
    def setNome(self,novoNome):
        self.__nome = novoNome
        
#Intanciação de um objeto da classe Mago
hp = Mago('Harry Potter', 17, 'Hogwarts')
gd = Mago('Gandalf', 2000, 'Magia Cinza')

#Acessando atributos públicos
# print(hp.__nome)
#hp.__nome = "Merlin"
hp.setNome("Merlin")

#Invocando métodos
hp.andar()
hp.falar()
hp.invocarMagia()
hp.cumprimentar("Rossana")

gd.falar()
gd.cumprimentar("Gabriel")

del hp
del gd